test/jdk/java/lang/module/ModuleDescriptorTest.java

```diff
@@ -1362,14 +1369,10 @@ public void testReadsWithPackageFinder() throws Exception {
      * complete set of packages.
      */
     public void testReadsWithBadPackageFinder() throws Exception {
-        ModuleDescriptor descriptor = ModuleDescriptor.newModule("foo")
-                .requires("java.base")
-                .exports("p")
-                .build();
-
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        ModuleInfoWriter.write(descriptor, baos);
-        ByteBuffer bb = ByteBuffer.wrap(baos.toByteArray());
+        ByteBuffer bb = ByteBuffer.wrap(Classfile.buildModule(
+                ModuleAttribute.of(ModuleDesc.of("foo"), mb -> mb
+                        .requires(ModuleDesc.of("java.base"), 0, null)
+                        .exports(PackageDesc.of("p"), 0))));
```

Comment:
If you split this up into two or third steps then it would be easier to read/debug.